src/Plugin/ContainerAwarePluginManager.php

```diff
@@ -59,10 +59,14 @@ public function hasDefinition($plugin_id) {
    * {@inheritdoc}
    */
   public function createInstance($plugin_id, array $configuration = array()) {
-    // @todo: Use ->expandArguments() when get() disallows getting private
-    //        services.
-    $plugin = clone $this->container->get($this->servicePrefix . $plugin_id);
-    return $plugin;
+    $plugin_definition = $this->getDefinition($plugin_id);
+    $plugin_definition += array(
+      'arguments' => array(),
+    );
+
+    array_unshift($plugin_definition['arguments'], $configuration);
+
+    return $this->container->createInstance($this->servicePrefix . $plugin_id, $configuration);
```

Comment:
I see the bug now, need to pass $plugin_definition and not $configuration here ...